x/crosschain/keeper/gas_price.go

```diff
@@ -6,36 +6,72 @@ import (
 	"cosmossdk.io/math"
 	"github.com/cosmos/cosmos-sdk/store/prefix"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"golang.org/x/exp/slices"
 
 	"github.com/zeta-chain/zetacore/x/crosschain/types"
 )
 
 // SetGasPrice set a specific gasPrice in the store from its index
 func (k Keeper) SetGasPrice(ctx sdk.Context, gasPrice types.GasPrice) {
 	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.GasPriceKey))
-	b := k.cdc.MustMarshal(&gasPrice)
 	gasPrice.Index = strconv.FormatInt(gasPrice.ChainId, 10)
+	b := k.cdc.MustMarshal(&gasPrice)
 	store.Set(types.KeyPrefix(gasPrice.Index), b)
 }
 
-// GetGasPrice returns a gasPrice from its index
-func (k Keeper) GetGasPrice(ctx sdk.Context, chainID int64) (val types.GasPrice, found bool) {
+// GetGasPrice returns a gasPrice from its index or false if it doesn't exist.
+func (k Keeper) GetGasPrice(ctx sdk.Context, chainID int64) (types.GasPrice, bool) {
 	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.GasPriceKey))
+
 	b := store.Get(types.KeyPrefix(strconv.FormatInt(chainID, 10)))
 	if b == nil {
-		return val, false
+		return types.GasPrice{}, false
 	}
+
+	var val types.GasPrice
 	k.cdc.MustUnmarshal(b, &val)
+
 	return val, true
 }
 
-func (k Keeper) GetMedianGasPriceInUint(ctx sdk.Context, chainID int64) (sdk.Uint, bool) {
-	gasPrice, isFound := k.GetGasPrice(ctx, chainID)
-	if !isFound {
-		return math.ZeroUint(), isFound
+// GetMedianGasPriceInUint returns the median gas price (and median priority fee) from the store
+// or false if it doesn't exist.
+func (k Keeper) GetMedianGasPriceInUint(ctx sdk.Context, chainID int64) (math.Uint, math.Uint, bool) {
```

Comment:
Maybe it can be renamed as it now returns priority fee as well?

Reply:
https://github.com/zeta-chain/node/pull/2483/commits/7c16387dd93d3863b09849169d6de3ead7c35b39